### DISCUSSION TITLE
Window "Pin" function to help rebuild standard work spaces...

### DISCUSSION THREAD

**Discussion by JerryWestrick** [pop-os/shell#1456] [Ideas]
### Situation
**In my standard work flow, I repeatedly find my self "moving" the same windows from/to the same places.** 

example:

On my 4k TV/screen I use the following lay out:

 1 2
 1 3

window 1 (full height and half of screen width) is my PyCharm python IDE
window 2 (Right top quarter of screen) is my web browser with google quick questions about the language / libraries.
window 3 (Right bottom quarter of screen) differs between terminal, HTOP, and other auxiliary tools being used at the time. 

##### The following scenario is repeated with every debug run:

1) In PyCharm I choose to start a debug run for Program "X"
2) the PyCharm Window is reduced to top, left quarter of the screen.
3) Program "X" is opened in the left bottom of screen
4) Using the mouse I move Program "X" to the Bottom Right of the screen,
     either on top of the Xterm (preferred) but often as it is so hard to drop the window into exact position 
     it ends up splitting the bottom right space with xterm.

### An Idea: "Pinning" window positions
If I could influence the placement algorithm, or even override it by "pinning" a certain window to it's current place, so the then next time the same window is created it is placed in the same position.  This will allow me to build and use work-spaces more effectively.

##### Workspace specific?
All of the bla bla sofar has been about a single workspace, my python development workspace, i.e. Work Space 3.  

- Workspace 1: Email / Calendar and other interruptions.
- Workspace 2: Office / and other necessary crap
- Workspace 3: Python Development 

### Foreseen problems / and solutions
The current placement algorithm is based on a Binary Tree type structure.  Although this does not really require any changes to the structure itself, the current placement algorithm is tightly tied to this concept.  This deviation would probably require a complete overhaul of the placement routines.  There are lots and lots of requests on the placement algorithms, maybe this concept could solve many of those?

**Comment by JerryWestrick** [comment]
Sorry, I forgot to give many many Kudus to the gals and guys who resurrected tiling window managers!!!   Awesome job and implementation

**Comment by bennypowers** [comment]
Would love to see this. For example, I'd always like tootle - the GTK mastodon client, to display as a thin tall window off to the side, whereas I want my browser and terminals to follow the regular tiling algorithm within the remaining space.
